src/ResourceManager/HDInsight/Commands.HDInsight/ManagementCommands/AddAzureHDInsightSecurityProfile.cs

```diff
@@ -0,0 +1,98 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Commands.HDInsight.Commands;
+using Microsoft.Azure.Commands.HDInsight.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.HDInsight
+{
+    [Cmdlet(
+        VerbsCommon.Add,
+        Constants.CommandNames.AzureHDInsightSecurityProfile),
+    OutputType(
+        typeof(AzureHDInsightSecurityProfile))]
+    public class AddAzureHDInsightSecurityProfile : HDInsightCmdletBase
+    {
+        private readonly AzureHDInsightSecurityProfile _securityProfile;
```

Comment:
It is making changes to the passed in configuration and so need to implement ShouldProcess.